keyserver/loader.mjs

```diff
@@ -15,7 +15,7 @@ async function resolve(specifier, context, nextResolve) {
       continue;
     }
     const path = localPackages[pkg];
-    const url = defaultResult.url.replace(path, `keyserver/dist/${pkg}`);
+    const url = defaultResult.url.replace(specifier, `keyserver/dist/${specifier}`);
```

Comment:
I find it really hard to believe that this works for `rust-node-addon`, and I don't understand how it solves your original issue either

Reply:
In following URL:
> file:///Users/XXXX/`web`3Service/comm/`web`/media/file-utils.js

where `path= 'web'` & `specifier= 'web/media/file-utils'`

previous logic was replacing **first** matching `path` with `keyserver/dist/${pkg}` which in my case (shared above URL) produced/resolved into the following incorrect URL:

> file:///Users/XXXX/`keyserver/dist/web`3Service/comm/web/media/file-utils.js`

THIS was my original scenario, which I've fixed by replacing `specifier` with `keyserver/dist/${specifier}` instead of `path` ( because there are very rare chances of `specifier`  duplications in a URL)

That's how given logic is resolving my issue. 
resolving this:
> file:///Users/XXXX/web3Service/comm/`web/media/file-utils`.js

into

> file:///Users/XXXX/web3Service/comm/`keyserver/dist/web/media/file-utils`.js

As far as `rust-node-addon` I can not produce it failing.

Reply:
That explanation does not make sense. You claim that "there are very rare chances of `specifier` duplication", but if you look at the code `specifier` is exactly equal to `pkg`, and `pkg` is exactly equal to `path` for all cases other than `rust-node-addon`. So your change results in exactly the same behavior for `web`, `landing`, and `lib`. Meanwhile, it's very likely to break `rust-node-addon`.

I suspect you aren't testing your changes very well.